src/libraries/System.Private.CoreLib/src/System/Text/EncodingTable.cs

```diff
@@ -119,6 +120,26 @@ internal static EncodingInfo[] GetEncodings()
             return arrayEncodingInfo;
         }
 
+        internal static EncodingInfo[] GetEncodings(Dictionary<int, EncodingInfo> encodingInfoList)
+        {
+            Debug.Assert(encodingInfoList != null);
+
+            for (int i = 0; i < s_mappedCodePages.Length; i++)
+            {
+                if (!encodingInfoList.TryGetValue(s_mappedCodePages[i], out _))
+                {
+                    encodingInfoList[s_mappedCodePages[i]] = new EncodingInfo(s_mappedCodePages[i], s_webNames[s_webNameIndices[i]..s_webNameIndices[i + 1]],
+                                                                              GetDisplayName(s_mappedCodePages[i], i));
+                }
+            }
+
+            var collection = encodingInfoList.Values;
+            EncodingInfo[] result = new EncodingInfo[collection.Count];
+            collection.CopyTo(result, 0);
```

Comment:
Not sure how much we care about efficiency here, but we could avoid a collection allocation by instead doing:
```C#
var result = new EncodingInfo[collection.Count];
int i = 0;
foreach (KeyValuePair<int, EncodingInfo> pair in encodingInfoList)
{
    result[i++] = pair.Value;
}
return result;
```